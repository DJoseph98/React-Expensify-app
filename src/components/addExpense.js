import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { addStartState } from '../actions/expenses';
import ExpenseForm from './expenseForm';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        //props.dispatch(addState(expense)); // difficile de tester comme ça donc on change
        this.props.addStartState(expense);// plus facile de tester la fonction maintenant
        this.props.history.push('/'); // redirect to route
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => { // surchage la fonction pour permettre de facilité le testing
    return {
        addStartState: (expense) => dispatch(addStartState(expense))
    }
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);