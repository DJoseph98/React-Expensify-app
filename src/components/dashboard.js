import React from 'react';
import ExpenseListe from './expenseList';
import ExpenseListFilter from './expenseListFilter';
import ExpenseSummary from './expenseSummary';
import SortBy from './sortFilter';

const DashboardPage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilter />
        <ExpenseListe />
    </div>
);

export default DashboardPage;