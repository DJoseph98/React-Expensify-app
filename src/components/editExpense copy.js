import React from 'react';
import ExpenseForm from './expenseForm';
import { connect } from 'react-redux';
import { editState, removeState } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.onSubmit(expense); //edit expense
        this.props.history.push('/'); // redirect to route
    }
    onClick = () => {
        this.props.onClick();
        //this.props.dispatch(removeState(({id: this.props.expense.id})));
        this.props.history.push('/'); // redirect to route
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} />
                <button onClick={this.onClick}>Remove</button>
            </div>)
    }
}

const MapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find(element => element.id === props.match.params.id) // retourne expense par id qu'on souhaite
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
        return {
            onSubmit: (expense) => dispatch(editState(expense.id, expense)),
            onClick: () => dispatch(removeState({ id: ownProps.id }))
        }
}

export default connect(MapStateToProps/*state*/, mapDispatchToProps/*dispatch*/)(EditExpensePage);