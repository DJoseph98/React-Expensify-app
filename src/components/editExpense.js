import React from 'react';
import ExpenseForm from './expenseForm';
import { connect } from 'react-redux';
import { startEditState, removeStartState } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditState(this.props.expense.id, expense); //edit expense
        this.props.history.push('/'); // redirect to route
    }
    onRemove = () => {
        this.props.removeStartState(({ id: this.props.expense.id }));
        this.props.history.push('/'); // redirect to route
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} />
                <button onClick={this.onRemove}>Remove</button>
            </div>)
    }
}

const MapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find(element => element.id === props.match.params.id) // retourne expense par id qu'on souhaite
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEditState: (id, expense) => dispatch(startEditState(id, expense)),
        removeStartState: (data) => dispatch(removeStartState(data))
    }
}

export default connect(MapStateToProps/*state*/, mapDispatchToProps/*dispatch*/)(EditExpensePage);