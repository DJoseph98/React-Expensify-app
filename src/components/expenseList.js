import { connect } from 'react-redux';
import React from 'react';
import ExpenseListeItem from './expenseListItem';
import filteredExpenses from '../selectors/expenses';

export const ExpenseListe = (props) => ( //component a render
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-dekstop">Expense</div>
            <div className="show-for-dekstop">Price</div>
        </div>
        <h2>Expenses List</h2>
        <div className="list-body">
            {props.expenses.length === 0 ? (
                <div className="list-item list-item--message">
                    <span>No expenses</span>
                </div>
            ) : (
                <u1>
                    {props.expenses.map(expense => <ExpenseListeItem key={expense.id}{...expense} />)}
                </u1>
            )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => { // maps les valeurs de state au variables du template
    return {
        expenses: filteredExpenses(state.expenses, state.filter)
    }
};

export default connect(mapStateToProps)(ExpenseListe); //connect les variables redux store avec les variables du template



