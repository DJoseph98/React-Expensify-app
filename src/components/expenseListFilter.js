import React from 'react';
import { connect } from 'react-redux';
import { sortByAmount, sortByDate, setStartDate, setEndDate, setTextFilter } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilter extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value); // set dans le store les data avec le tri qu'on veux
    }
    onSortByChange = (e) => {
        switch (e.target.value) {
            case 'date':
                this.props.sortByDate();
            case 'price':
                this.props.sortByAmount();
        }
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type="text" className="text-input" placeholder="Search expenses" value={this.props.filter.text} onChange={this.onTextChange}></input>
                    </div>
                    <div className="input-group__item">
                        <select className="select" value={this.props.filter.sortBy} onChange={this.onSortByChange}>
                            <option value="date">Date</option>
                            <option value="price">Price</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filter.startDate}
                            endDate={this.props.filter.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                        />
                    </div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        filter: state.filter
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate)),
        setTextFilter: (text) => dispatch((setTextFilter(text))),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount()),
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilter);