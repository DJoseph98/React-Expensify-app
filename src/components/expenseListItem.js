import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removeState } from '../actions/expenses';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListeItem = ({ dispatch, id, description, price, createdAt }) => ( //component a render
        <Link className="list-item" to={`/edit/${id}`}>
                <div>
                        <h3 className="list-item__title">{description}</h3>
                        <span className="list-item__sub-title">{moment(createdAt).format("MM-DD-YYYY")}</span>
                </div>
                <h3 className="list-item__data">{numeral(price).format('$0,0.00')}</h3>
        </Link>
);

export default connect()(ExpenseListeItem);


