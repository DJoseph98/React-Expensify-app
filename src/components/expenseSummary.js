import React from 'react';
import { connect } from 'react-redux';
import getTotalExpense from '../selectors/expense-total';
import filteredExpenses from '../selectors/expenses';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">
                Total expenses : <span>{expenseCount}</span>

                {expenseTotal !== 0 && <div>Total amount expenses : <span>{numeral(expenseTotal).format('$0,0.00')}</span> </div>}
            </h1>

            <div className="page-header__actions">
                <Link className="button" to="/create">Add expense</Link>
            </div>
        </div>
    </div>
)

const mapStateToProps = (state) => { // maps les valeurs de state au variables du template
    const visibleExpenses = filteredExpenses(state.expenses, state.filter);

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getTotalExpense(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);