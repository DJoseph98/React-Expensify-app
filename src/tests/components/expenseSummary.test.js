import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/expenseSummary';

test('viewing 2 expenses list', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={0} expenseTotal={10} />);
    expect(wrapper).toMatchSnapshot();
});

test('viewing 3 expenses list', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expenseTotal={1051515} />);
    expect(wrapper).toMatchSnapshot();
});