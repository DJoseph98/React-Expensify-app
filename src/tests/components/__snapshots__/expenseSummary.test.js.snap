// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewing 2 expenses list 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Total expenses : 
      <span>
        0
      </span>
      <div>
        Total amount expenses : 
        <span>
          $10.00
        </span>
         
      </div>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`viewing 3 expenses list 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Total expenses : 
      <span>
        23
      </span>
      <div>
        Total amount expenses : 
        <span>
          $1,051,515.00
        </span>
         
      </div>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;
